### ISSUE TITLE
onReceived buff 最长64位吗?

### ISSUE THREAD

**Issue by WangShayne** [bastengao/react-native-serial-port-api#52]
观察到每次打印收到64位长度的数组

**Comment by bastengao** [comment]
是的，用 64 长度的数组去读数据。

**Comment by TheGlenn88** [comment]
@bastengao is there a reason it's fixed at 64 bytes?

**Comment by bastengao** [comment]
> @bastengao is there a reason it's fixed at 64 bytes?

 It's buffer size and must have a fixed size. buffer size could be any length, I just choose 64.

Serial port sends data stream, it's endless. But it must be read by fixed buffer. Application layer must make a protocol that could split packet in data stream.

**Comment by TheGlenn88** [comment]
I've been looking at the original android implementation and can see a couple of forks, this one is based on https://github.com/licheedev/Android-SerialPort-API (i think)

There is another that talks about the 64byte issue and i think solves it by reading the available bytes in the stream.
see here: https://github.com/xmaihh/Android-Serialport/issues/14

is that not something that can be done here?

**Comment by TheGlenn88** [comment]
It's currently impossible to know if the data received is the end of the data.

For example if the source device sent data that was 140 bytes, I could do something like

I've received 64 bytes, lets wait for the next packet...
i've received 64 bytes, lets wait for the next packet...
i've received 12 bytes, this must be the end? **process**

but if the device is sending exactly 128 byes, thats a different problem, you could do

I've received 64 bytes, lets wait for the next packet...
i've received 64 bytes, lets wait for the next packet...
timeout 500ms... **process**

if another packet unrelated is sent in the timeout period, it can't be processed reliably

It would require the device sending the data to our React Native app to send payload size data. If you don't control how the source device sends it's data then you can't reliably put the data packets together.
I hope that makes sense, perhaps there is something i'm missing?

**Comment by bastengao** [comment]
@TheGlenn88 What can i help you?

**Comment by WangShayne** [comment]
@TheGlenn88 
Received data must contain closing tag,i think

@bastengao 
Perhaps it can be more flexible, such as default 64, but can be customized?

**Comment by TheGlenn88** [comment]
@bastengao @WangShayne I've been having a go at creating a NativeModule and can now see the issue with the `InputStream`. The best way would be to know a definitive start and end once there is some data in the stream.

I've also found that the other library reads `available`, but was still returning 64 bytes. It just happens by chance, that at 9600 baud rate, it equates to (1200 bytes / 1000ms) * 50ms = 60 bytes per 50ms, give or take a few cycles and it was landing near 64bytes, sometimes more, sometimes less.

I've also found that by upping the baud rate I am able to get more data through in the 50ms wait between empty input streams.

It seems not a lot can be done if the data is not sent with the packet size data.
